src/main/java/strategy/CalculateStrategy.java

```diff
@@ -0,0 +1,48 @@
+package strategy;
+
+import db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class CalculateStrategy {
+    private final Storage storage;
+    private final Map<String, Integer> fruitQuantities;
+
+    public CalculateStrategy(Storage storage, Map<String, Integer> fruitQuantities) {
+        this.storage = storage;
+        this.fruitQuantities = fruitQuantities;
+    }
+
+    public void setOperation(FruitTransaction fruitTransaction) {
+        int storageValue = 0;
+
+        switch (fruitTransaction.getOperation()) {
+            case BALANCE -> {
+                storage.put(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+                break;
+            }
+            case SUPPLY, RETURN -> {
+                Integer currentQuantity = storage.get(fruitTransaction.getFruit());
+                if (currentQuantity != null) {
+                    storageValue = currentQuantity;
+                }
+                storage.put(fruitTransaction.getFruit(),
+                        storageValue + fruitTransaction.getQuantity());
+                break;
+            }
+            case PURCHASE -> {
+                Integer currentQuantity = storage.get(fruitTransaction.getFruit());
+                if (currentQuantity != null) {
+                    storageValue = currentQuantity;
+                }
+                storage.put(fruitTransaction.getFruit(),
+                        storageValue - fruitTransaction.getQuantity());
+                break;
+            }
+            default -> throw new IllegalArgumentException("Unexpected operation: "
+                        + fruitTransaction.getOperation());
+        }
+        fruitQuantities.put(fruitTransaction.getFruit(),
+                storage.get(fruitTransaction.getFruit()));
+    }
+}
```

Comment:
strategy pattern is not used here
should be something like
```suggestion
public class OperationStrategy {
    private final Map<FruitTransaction.Operation, OperationHandler> handlerByOperation;

    public OperationStrategy(Map<FruitTransaction.Operation, OperationHandler> handlerByOperation) {
        this.handlerByOperation = handlerByOperation;
    }

    public OperationHandler getHandler(FruitTransaction.Operation operation) {
        return handlerByOperation.get(operation);
    }
}
```
- map is coming as argument from Main method during initialization
- strategy goal is provide implementation proper of handler to operation
- logic that work with storage should be moved to handlers
- handler should have one method like `void processTransaction()` that will change state of storage properly